rclcpp/src/rclcpp/node_interfaces/node_graph.cpp

```diff
@@ -205,13 +205,13 @@ NodeGraph::get_graph_event()
 {
   auto event = rclcpp::event::Event::make_shared();
   std::lock_guard<std::mutex> graph_changed_lock(graph_mutex_);
+  graph_events_.push_back(event);
+  graph_users_count_++;
   // on first call, add node to graph_listener_
   if (should_add_to_graph_listener_.exchange(false)) {
     graph_listener_->add_node(this);
     graph_listener_->start_if_not_started();
   }
-  graph_events_.push_back(event);
-  graph_users_count_++;
```

Comment:
This block was moved because there was a data race between adding this node to the `graph_listener_` and updating the state internal to the node. This was just a silly mistake on my part.